canonical-webteam/practices
# Automate navigation in website

Here I'm doing quite a few things:

- Move the title of each page into metadata (`title: {title}`)
- Move the description of each page (from `index.md` that was) into metadata (`description: {description}`)
- Generate the side-nav automatically, based on filenames
- Generate the list in the index page automatically, based on `title` and `description`
- Put the page title in its own fancy row, with a nice background
- Change the name of the site to "Web and design practices" (hopefully fixes #117)
- Add `jekyll-redirect-from` ([the only plugin supported by GitHub pages](https://help.github.com/articles/redirects-on-github-pages/)), and redirect `/coding/react` and `/coding/code-formatting` to `/coding/javascript`

QA
--

``` bash
sudo gem install jekyll-redirect-from
jekyll serve
```

Go to http://localhost:4000/practices/ and check the site.

Go to http://localhost:4000/practices/coding/react and http://localhost:4000/practices/coding/code-formatting and check you are redirected to http://localhost:4000/practices/coding/javascript.html.
